### ISSUE TITLE
Tilde expansion in REPL shell completion raises an error

### ISSUE THREAD

**Issue by dhoegh** [JuliaLang/julia#11281]
When using tilde expansion then tabbing just after the tilde causes an error to be raised.
```
julia> "~<tab>
julia> "~WARNING: Caught an exception in the keymap:
ERROR: BoundsError
ERROR: syntax: incomplete: invalid string syntax
```
Another issue is the regex on https://github.com/JuliaLang/julia/blob/8d3a42b62d21ed54909426bb8dc77ef463ccd4d3/base/REPLCompletions.jl#L114 is not windows compatible. So using back slashes on windows do not work
```
julia> "~\\<tab>
```
This was introduced in #7617 
CC @garborg

**Comment by blakejohnson** [comment]
I see what the problem is now. I'll have to think about a slight generalization to make it work on windows.

**Comment by blakejohnson** [comment]
Fixed in 39e83f7
